packages/react/src/components/DetailsList/DetailsHeader.base.tsx

```diff
@@ -140,6 +140,11 @@ export class DetailsHeaderBase
     if (this.props.isAllCollapsed !== prevProps.isAllCollapsed) {
       this.setState({ isAllCollapsed: this.props.isAllCollapsed });
     }
+
```

Comment:
If this state is input by the host app/component, then there's no need to call `onEditColumn` since the host already knows what's up.

It seems the issue here is you just need a way to obtain the `updateColumn` function from the `DetailsList`, so you can call it ad-nauseum from whatever experience you want to create. But there isn't a need for `DetailsHeader` to keep track of much.